src/coreclr/jit/compiler.h

```diff
@@ -9106,8 +9112,25 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         {
             return MinOpts() || compDbgCode;
         }
-        bool OptimizationEnabled() const
+
+        //------------------------------------------------------------------------
+        // OptimizationEnabled: Are optimizations of the specified kind allowed?
+        //
+        // Arguments:
+        //    kind - there are two kinds of optimizations:
+        //      OPT_All:         all kinds of optimizations
+        //      OPT_Lightweight: only those which won't regress JIT throughput
+        //
+        // Return Value:
+        //    true if JIT is alowed to perform optimizations of the given kind.
+        //
+        bool OptimizationEnabled(OptimizationsKind kind = OPT_All) const
         {
+            if (kind == OPT_Lightweight)
+            {
+                // Quick optimizations are allowed for all cases except debug code mode
```

Comment:
I'd think we'd still want to force-disable all optimizations under explicit minopts.

Reply:
Yeah, I was thinking about that too, but looks like currently our `opts.MinOpts()` just returns true for any Tier0 🤔

Reply:
Ah, `jitFlags->IsSet(JitFlags::JIT_FLAG_MIN_OPT)`  should do the trick

Reply:
Yes, since now we only have two "optimization levels". So it'd require some more changes to add an in-between one.

Reply:
Yep, initially I planned to refactor `MinOpts()` but that's a lot of changes which would mask these changes so hopefully in a separate PR.